lib/dictBuilder/cover.c

```diff
@@ -620,14 +644,24 @@ static size_t COVER_ctx_init(COVER_ctx_t *ctx, const void *samplesBuffer,
     for (i = 0; i < ctx->suffixSize; ++i) {
       ctx->suffix[i] = i;
     }
-    /* qsort doesn't take an opaque pointer, so pass as a global.
-     * On OpenBSD qsort() is not guaranteed to be stable, their mergesort() is.
-     */
-    g_coverCtx = ctx;
-#if defined(__OpenBSD__)
+#if defined(__APPLE__)
```

Comment:
`qsort_r()` is invoked directly from within in `COVER_ctx_init()`, resulting in some `#if ... #elif ... #else ... ` preprocessor logic in the function's body, because `qsort_r()` is different depending on target system.
In a bid to lower preprocessor logic within `COVER_ctx_init()`, I was wondering if this design could be simplified by an invocation to a single function signature, and this function would be implemented differently depending on target system. 
Such a design requires a common signature across all systems, so there might be some limits there.